### PR TITLE
chore(dashmate): update Core to version 22

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Update Platform to Core 22

## What was done?

- Updated Core to v22.0.0-rc.1
- Added config migration
- Fixed status command
- Fixed the local network setup

## How Has This Been Tested?

None

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Introduced a new migration function for version `1.6.0` to update Docker image configurations.
	- Added a "Create wallet" task and a task for generating collateral funds for masternodes.
	- Updated activation tasks for DIP3 and V21 fork with new parameters.

- **Improvements**
	- Simplified platform activation status reporting.
	- Enhanced error handling for Docker status checks.

- **Bug Fixes**
	- Updated Docker image version to `dashpay/dashd:22.0.0-rc.1`.

### CODE REVIEW

**Comment by QuantumExplorer** [comment]
Do not merge this!

**Comment by shumkov** [comment]
@QuantumExplorer what about now?

**Comment by QuantumExplorer** [comment]
Let's make this for 1.4 only please.

**Comment by QuantumExplorer** [comment]
We can't really merge this into production.

**Comment by shumkov** [comment]
https://github.com/dashpay/dash/pull/6247

**Review by shumkov — COMMENTED** [review]
chore: update the image to latest nightly

File: packages/dashmate/src/listr/tasks/setup/local/configureCoreTaskFactory.js [hunk 1]
```diff
@@ -93,9 +92,40 @@ export default function configureCoreTaskFactory(
...
+            {
+              title: 'Generating funds to use as a collateral for masternodes',
+              task: () => {
+                const amount = HPMN_COLLATERAL_AMOUNT * configGroup.length;
+                return generateToAddressTask(
+                  configGroup.find((c) => c.getName() === 'local_seed'),
+                  amount,
+                );
+              },
+            },
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/setup/local/configureCoreTaskFactory.js:124]: Don't worry about it. It's adding above.

File: packages/dashmate/templates/core/dash.conf.dot [hunk 2]
```diff
@@ -93,6 +92,7 @@ fallbackfee=0.00001
 {{?? it.network === 'local'}}
 regtest=1
 [regtest]
+testactivationheight=mn_rr@1000
```
Comment (QuantumExplorer) [inline packages/dashmate/templates/core/dash.conf.dot:95]: Is this correct?

**Review by QuantumExplorer — DISMISSED** [review]

File: packages/dashmate/configs/defaults/getBaseConfigFactory.js [hunk 3]
```diff
@@ -53,7 +53,7 @@ export default function getBaseConfigFactory() {
           port: 3001,
         },
         docker: {
-          image: 'dashpay/dashd:21',
+          image: 'dashpay/dashd:22.0.0-rc.1',
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/configs/defaults/getBaseConfigFactory.js:56]: No worries. We will change to 22 when it will be released.

**Pull request merged**